File: ironpulse-frontend/src/types/payment.ts
// ── Payment types ────────────────────────────────────

export interface VipStatus {
  status: 'active' | 'cancelled' | 'expired' | 'trial' | 'none'
  plan: string | null
  started_at: string | null
  expires_at: string | null
  auto_renew: boolean
  price_myr: number | null
}

export interface PaymentInitiateRequest {
  plan: 'monthly' | 'annual'
  payment_method: 'fpx' | 'tng' | 'boost' | 'card'
  return_url: string
  gateway?: 'billplz' | 'toyyibpay'
}

export interface PaymentInitiateResponse {
  payment_id: string
  payment_url: string
  amount_myr: number
  sst_myr: number
  total_myr: number
}

export interface PaymentRecord {
  id: string
  user_id: string
  amount_myr: number
  payment_method: string
  provider: string
  provider_payment_id: string | null
  status: string
  plan: string
  metadata_json: Record<string, unknown> | null
  created_at: string
  completed_at: string | null
}

export interface PaymentVerify {
  payment_id: string
  status: string
  paid: boolean
  completed_at: string | null
}

export interface Gym {
  id: string
  name: string
  address: string | null
  city: string | null
  state: string | null
  latitude: number | null
  longitude: number | null
  phone: string | null
  hours: Record<string, unknown> | null
  is_24h: boolean
  is_vip_partner: boolean
  features: string[] | null
  photo_urls: string[] | null
  rating: number | null
  review_count: number
  distance_km?: number
}

File: ironpulse-backend/app/schemas/payment.py
"""
Payment / VIP schemas — updated with SST, initiation, verification.
"""

import datetime
from pydantic import BaseModel


# ── VIP ───────────────────────────────────────────────
class VipStatusOut(BaseModel):
    status: str  # active | cancelled | expired | trial | none
    plan: str | None = None
    started_at: datetime.datetime | None = None
    expires_at: datetime.datetime | None = None
    auto_renew: bool = False
    price_myr: float | None = None


class VipSubscribeRequest(BaseModel):
    plan: str  # monthly | annual
    payment_method: str = "fpx"  # fpx | tng | boost | card
    return_url: str


class VipCancelResponse(BaseModel):
    message: str
    status: str
    auto_renew: bool


# ── Payment initiation ───────────────────────────────
class PaymentInitiateRequest(BaseModel):
    plan: str  # monthly | annual
    payment_method: str = "fpx"  # fpx | tng | boost | card
    return_url: str
    gateway: str = "billplz"  # billplz | toyyibpay


class PaymentInitiateResponse(BaseModel):
    payment_id: str
    payment_url: str
    amount_myr: float
    sst_myr: float
    total_myr: float


# ── Payment record ────────────────────────────────────
class PaymentOut(BaseModel):
    id: str
    user_id: str
    amount_myr: float
    payment_method: str
    provider: str
    provider_payment_id: str | None = None
    status: str
    plan: str
    metadata_json: dict | None = None
    created_at: datetime.datetime
    completed_at: datetime.datetime | None = None

    model_config = {"from_attributes": True}


class PaymentVerifyOut(BaseModel):
    payment_id: str
    status: str
    paid: bool
    completed_at: datetime.datetime | None = None
